test/v2/controllers/requestParsers/validators/CreateForeignPropertyPeriodSummaryValidatorSpec.scala

```diff
@@ -0,0 +1,2085 @@
+        validator.validate(CreateForeignPropertyPeriodSummaryRawData(validNino, validBusinessId, validTaxYear, Json.parse(
+          """
+            |{
+            |  "fromDate": "2020-01-01",
+            |  "toDate": "2020-01-31",
+            |  "foreignFhlEea": {
+            |    "income": {
+            |      "rentAmount": 567.83
+            |      },
+            |    "expenses": {
+            |      "premisesRunningCosts": 4567.98,
+            |      "repairsAndMaintenance": 98765.67,
+            |      "financialCosts": 4566.95,
+            |      "professionalFees": 23.65,
+            |      "costsOfServices": 4567.77,
+            |      "travelCosts": 456.77,
+            |      "other": 567.67
+            |    }
+            |
+            |  },
+            |  "foreignNonFhlProperty": [
+            |    {
+            |      "countryCode": "12345678",
+            |      "income": {
+            |        "rentIncome": {
+            |          "rentAmount": 34456.30
+            |        },
+            |        "foreignTaxCreditRelief": true,
+            |        "premiumsOfLeaseGrant": 2543.43,
+            |        "otherPropertyIncome": 54325.30,
+            |        "foreignTaxTakenOff": 6543.01,
+            |        "specialWithholdingTaxOrUKTaxPaid": 643245.00
+            |      },
+            |      "expenses": {
+            |        "premisesRunningCosts": 5635.43,
+            |        "repairsAndMaintenance": 3456.65,
+            |        "financialCosts": 34532.21,
+            |        "professionalFees": 32465.32,
+            |        "costsOfServices": 2567.21,
+            |        "travelCosts": 2345.76,
+            |        "other": 2425.11
+            |      }
+            |    },
+            |    {
+            |      "countryCode": "34567890",
+            |      "income": {
+            |        "rentIncome": {
+            |          "rentAmount": 34456.30
+            |        },
+            |        "foreignTaxCreditRelief": true,
+            |        "premiumsOfLeaseGrant": 2543.43,
+            |        "otherPropertyIncome": 54325.30,
+            |        "foreignTaxTakenOff": 6543.01,
+            |        "specialWithholdingTaxOrUKTaxPaid": 643245.00
+            |      },
+            |      "expenses": {
+            |        "premisesRunningCosts": 5635.43,
+            |        "repairsAndMaintenance": 3456.65,
+            |        "financialCosts": 34532.21,
+            |        "professionalFees": 32465.32,
+            |        "costsOfServices": 2567.21,
+            |        "travelCosts": 2345.76,
+            |        "other": 2425.11
+            |      }
+            |    }
+            |  ]
+            |}
+            |""".stripMargin
+        ))) shouldBe List(CountryCodeFormatError.copy(paths = Some(Seq("/foreignNonFhlProperty/0/countryCode", "/foreignNonFhlProperty/1/countryCode"))))
+      }
+    }
+
```

Comment:
Add tests for duplicate country code detection (recently added to the spec). See [AmendForeignPropertyPeriodSummaryValidator#L71](https://github.com/hmrc/property-business-api/blob/d645b7f47057179151e176825b7195583ece0e46/app/v2/controllers/requestParsers/validators/AmendForeignPropertyPeriodSummaryValidator.scala#L71) and [AmendForeignPropertyPeriodSummaryValidatorSpec#L429](https://github.com/hmrc/property-business-api/blob/d645b7f47057179151e176825b7195583ece0e46/test/v2/controllers/requestParsers/validators/AmendForeignPropertyPeriodSummaryValidatorSpec.scala#L429) (both currently under review :grimacing:)